### DISCUSSION TITLE
CORS errors on Chrome?

### DISCUSSION THREAD

**Discussion by DCS17** [mcguirepr89/BirdNET-Pi#565] [Q&A] [answered]
I've started getting a problem in Chrome browser since upgrade to OSX 12.6.
Causing web pages to fail to render properly. (see screenshot).
Safari is fine, but can't use the scrolling Spectrogram which I like to watch :-)

The Chrome console is showing multiple errors eg.
_Access to CSS stylesheet at 'http://birdnetpi.local/style.css?v=8.05.22' from origin 'http://birdnetpi.local' has been blocked by CORS policy: The request client is not a secure context and the resource is in more-private address space `private`._

Not sure what's happened or how to fix? 
Googling 'CORS policy' has simply confused me and before I totally brick my installation which is working perfectly on Safari, I thought I'd ask for some knowledgable help!
Any help much appreciated. Thanks

![Screenshot](https://user-images.githubusercontent.com/64783730/191924653-4af4f5e1-d076-4e7c-94a7-a95736a05761.jpg)

**Comment by mcguirepr89** [comment, marked as answer]
I don't have access to an OSX system, so could you please provide the Chrome browser version number?

So far, I've not been able to replicate this and am hoping someone else might be able to replicate this. 

I have a really limited understanding of CORS, but may I ask if you are trying to access the site via `http://birdnetpi.local` in the address bar? (That seems to be part of the reply header and would want to know what the request looked like)

**Reply by DCS17** [reply]
Chrome is up to date
Version 105.0.5195.125 (Official Build) (arm64)
Yes the site is accessed with  http://birdnetpi.local
I've tried all the usual of deleting caches / rebooting, but no change. I've tried chrome site settings to allow insecure content as the browser bar warns that site is insecure (obviously as http:). But that hasn't helped either. All a bit odd !

**Reply by DCS17** [reply]
I've been trying to give myself a crash course in website development and caddy! No prior experience at all!

I've no idea whether this is approriate or not, but I've removed the http:// from the localhost and birdpinet.local in the caddyfile. I think this forces https? Anyway, now Chrome and Safari complain the site is not secure, but when I ignore the warnings, then it seems to run fine on both broswers. 

As I only use the site for my local use (I can't publish websites easily because I am on 4G for internet with CGNAT) then it's a solution that I'm happy with.

**Reply by ehpersonal38** [reply]
Yeah this one seems very strange to me, especially that it only occurs on Mac and not Windows or Linux. The message doesn't make sense to me since CORS should only apply to cross site stuff. I.e. http://mycoolsite.com trying to pull in a script or stylesheet from http://birdnetpi.local. But you're seeing this message when http://birdnetpi.local is trying to load resources from itself. Don't make no sense to me. hah. Glad you arrived at a solution though!
